Mathlib/CategoryTheory/Limits/Preserves/Bifunctor.lean

```diff
@@ -0,0 +1,356 @@
+  isColimit :=
+    PreservesColimit₂.isColimit_mapCocone₂
+      c₁.isColimit
+      c₂.isColimit|>.some
+
+variable [HasColimit K₁] [HasColimit K₂]
+
+/-- Get a colimit cocone over `(whiskeringLeft₂ C|>.obj K₁|>.obj K₂|>.obj G)` provided the
+relevant colimits exist. This colimit cocone has definitionally
+`(G.obj <| colimit K₁).obj (colimit K₂)` as its cocone point. -/
+noncomputable def getColimitCocone₂ :
+    ColimitCocone <| uncurry.obj (whiskeringLeft₂ C|>.obj K₁|>.obj K₂|>.obj G) :=
+  mkColimitCocone₂ G (getColimitCocone K₁) (getColimitCocone K₂)
+
+-- The simp forms below are preferred over the ones generated by @[simps].
+
+@[simp]
+lemma getColimitCocone₂_cocone_pt :
+    (getColimitCocone₂ K₁ K₂ G).cocone.pt = (G.obj <| colimit K₁).obj (colimit K₂) :=
+  rfl
+
+@[simp]
+lemma getColimitCocone₂_cocone_ι_app (j : J₁ × J₂) :
+    (getColimitCocone₂ K₁ K₂ G).cocone.ι.app j =
+    (G.map <| colimit.ι K₁ j.1).app (K₂.obj j.2) ≫ (G.obj <| colimit K₁).map (colimit.ι K₂ j.2) :=
+  rfl
+
+/-- Extract the isomorphism between
+`colim (uncurry.obj (whiskeringLeft₂ C|>.obj K₁|>.obj K₂|>.obj G))` and
+`(G.obj (colim K₁)).obj (colim K₂)` from a `PreservesColimit₂` instance, provided the relevant
+colimits exist. -/
+noncomputable def isoColimitUncurryWhiskeringLeft₂ :
+    colimit (uncurry.obj (whiskeringLeft₂ C|>.obj K₁|>.obj K₂|>.obj G)) ≅
+    (G.obj <| colimit K₁).obj (colimit K₂) :=
+  IsColimit.coconePointUniqueUpToIso (colimit.isColimit _) (getColimitCocone₂ K₁ K₂ G).isColimit
+
+/-- Characterize the forward direction of the isomorphism
+`PreservesColimit₂.isoColimitUncurryWhiskeringLeft₂` w.r.t the canonical maps to the colimit. -/
+@[reassoc (attr := simp)]
+lemma ι_comp_isoColimitUncurryWhiskeringLeft₂_hom (j : J₁ × J₂) :
+    colimit.ι (uncurry.obj (whiskeringLeft₂ C|>.obj K₁|>.obj K₂|>.obj G)) j ≫
+      (PreservesColimit₂.isoColimitUncurryWhiskeringLeft₂ K₁ K₂ G).hom =
+    (G.map <| colimit.ι K₁ j.1).app _ ≫ (G.obj _).map (colimit.ι K₂ j.2) := by
+  dsimp [PreservesColimit₂.isoColimitUncurryWhiskeringLeft₂, Functor.mapCocone₂]
+  aesop_cat
+
+/-- Characterize the forward direction of the isomorphism
+`PreservesColimit₂.isoColimitUncurryWhiskeringLeft₂` w.r.t the canonical maps to the colimit. -/
+@[reassoc (attr := simp)]
+lemma map_ι_comp_isoColimitUncurryWhiskeringLeft₂_inv (j : J₁ × J₂) :
+    (G.map (colimit.ι K₁ j.1)).app _ ≫ (G.obj _).map (colimit.ι K₂ j.2) ≫
+      (PreservesColimit₂.isoColimitUncurryWhiskeringLeft₂ K₁ K₂ G).inv =
+    colimit.ι (uncurry.obj (whiskeringLeft₂ C|>.obj K₁|>.obj K₂|>.obj G)) j := by
+  rw [← Category.assoc, Iso.comp_inv_eq]
+  simp
+
+end
+
+/-- If a bifunctor preserves separately colimits of `K₁` in the first variable and colimits
+of `K₂` in the second variable, then it preserves colimit of the pair `K₁, K₂`. -/
+instance of_preservesColimits_in_each_variable [HasColimitsOfShape J₁ C]
+    [h₀ : PreservesColimit K₁ G] [∀ x : C₁, PreservesColimit K₂ (G.obj x)] :
```

Comment:
Instead of `PreservesColimit K₁ G`, would not it be more symmetric to use `∀ x, PreservesColimit K₁ (G.flip.obj x)`?